operators/knative-istio-controller/src/charm.py

```diff
@@ -62,32 +65,68 @@ def render(self):
             verbs=None,
         )
 
-        # FIXME: remove hardcoded knative-serving 
+        # FIXME: remove hardcoded knative-serving
         args = {
-            "gateway_name": "knative-custom-gateway",
-            "gateway_namespace": self.model.name,
             "name": "istio-controller",
             "namespace": self.model.name,
-            "knative_serving": "knative-serving"
+            "knative_serving": "knative-serving",
         }
 
         templates = [
-            env.get_template("deployment.yaml.j2").render(**args),
-            env.get_template("config.yaml.j2").render(**args),
-            env.get_template("rbac.yaml.j2").render(**args),
+            self.env.get_template("deployment.yaml.j2").render(**args),
+            self.render_config_istio(args),
```

Comment:
I think I'd organise how istio's args get in here a bit differently.  It doesn't feel intuitive that `self.render_config_istio(args)` is going to mutate `args`, adding the gateway name/namespace into the dict. 

Maybe something more like
```python
gateway_name, gateway_namespace = self.get_gateway_config()
args = {
  "gateway_name": gateway_name,
  ...
}
```
?